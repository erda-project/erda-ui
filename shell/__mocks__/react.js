// Copyright (c) 2021 Terminus, Inc.
//
// This program is free software: you can use, redistribute, and/or modify
// it under the terms of the GNU Affero General Public License, version 3
// or later ("AGPL"), as published by the Free Software Foundation.
//
// This program is distributed in the hope that it will be useful, but WITHOUT
// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
// FITNESS FOR A PARTICULAR PURPOSE.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program. If not, see <http://www.gnu.org/licenses/>.

const react = require('react');
// Resolution for requestAnimationFrame not supported in jest error :
// https://github.com/facebook/react/issues/9102#issuecomment-283873039
global.window = global;
window.addEventListener = () => {};
window.requestAnimationFrame = () => {
  // throw new Error('requestAnimationFrame is not supported in Node');
};

module.exports = react;
