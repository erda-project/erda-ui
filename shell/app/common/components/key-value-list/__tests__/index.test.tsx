// Copyright (c) 2021 Terminus, Inc.
//
// This program is free software: you can use, redistribute, and/or modify
// it under the terms of the GNU Affero General Public License, version 3
// or later ("AGPL"), as published by the Free Software Foundation.
//
// This program is distributed in the hope that it will be useful, but WITHOUT
// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
// FITNESS FOR A PARTICULAR PURPOSE.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program. If not, see <http://www.gnu.org/licenses/>.

import React from 'react';
import { render } from '@testing-library/react';
import KeyValueList from '..';

jest.mock('common/components/markdown-render', () => {
  // eslint-disable-next-line @typescript-eslint/no-shadow
  const React = require('react');
  const MarkdownRender = (props) => {
    return <div className="mock-markdown-render">{props.value}</div>;
  };
  return MarkdownRender;
});

describe('KeyValueList', () => {
  afterAll(() => {
    jest.resetAllMocks();
  });
  const data = {
    name: 'erda-development',
    envs: ['test', 'dev', 'staging', 'pros'],
    activeTime: '2022-03-16 16:05:30',
    workSpace: {
      readme: '### this is markdown',
    },
    project: [
      {
        name: 'erda',
        env: {
          test: {
            envName: 'test',
          },
          dev: {
            envName: 'develop',
          },
        },
        application: [
          {
            name: 'app1',
          },
        ],
      },
    ],
    description: 'this is description',
  };
  it('should render well', () => {
    const result = render(<KeyValueList data={data} markdownTextFields={['readme']} />);
    expect(result.container).toMatchSnapshot();
    expect(result.container).not.isExistClass('.key-value-list', 'shrink');
    result.rerender(<KeyValueList data={data} markdownTextFields={['readme']} shrink />);
    expect(result.container).isExistClass('.key-value-list', 'shrink');
    result.rerender(
      <KeyValueList
        data={data}
        markdownTextFields={['readme']}
        shrink
        listRender={(v) => {
          return <div className="custom-list-render">custom-list-render: {JSON.stringify(v)}</div>;
        }}
        textRender={(k, v) => {
          return (
            <div className="custom-text-render">
              custom-text-render: {k}-{v}
            </div>
          );
        }}
      />,
    );
    expect(result.queryAllByText(/custom-list-render/).length).toBeGreaterThan(0);
    expect(result.queryAllByText(/custom-text-render/).length).toBeGreaterThan(0);
  });
});
