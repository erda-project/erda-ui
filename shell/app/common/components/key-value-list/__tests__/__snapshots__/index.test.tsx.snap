// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyValueList should render well 1`] = `
<div>
  <div
    class="key-value-list depth-1  "
  >
    <div
      class="k-v-row"
    >
      <span
        class="key"
      >
        name
      </span>
      <span
        class="value"
      >
         
        erda-development
         
      </span>
    </div>
    <div
      class="k-v-row"
    >
      <span
        class="key"
      >
        envs
      </span>
      <span
        class="value"
      >
         
        test , dev , staging , pros
         
      </span>
    </div>
    <div
      class="k-v-row"
    >
      <span
        class="key"
      >
        activeTime
      </span>
      <span
        class="value"
      >
         
        2022-03-16 16:05:30
         
      </span>
    </div>
    <div
      class="k-v-row"
    >
      <span
        class="key"
      >
        readme
      </span>
      <span
        class="w-7/10 overflow-auto max-h-72"
      >
        <div
          class="mock-markdown-render"
        >
          ### this is markdown
        </div>
      </span>
    </div>
    <div
      class="key-value-list depth-2 sub "
    >
      <div
        class="title"
      >
        project
      </div>
      <div
        class="k-v-row"
      >
        <span
          class="key"
        >
          name
        </span>
        <span
          class="value"
        >
           
          erda
           
        </span>
      </div>
      <div
        class="key-value-list depth-3 sub "
      >
        <div
          class="title"
        >
          env
        </div>
        <div
          class="k-v-row"
        >
          <span
            class="key"
          >
            envName
          </span>
          <span
            class="value"
          >
             
            test
             
          </span>
        </div>
        <div
          class="k-v-row"
        >
          <span
            class="key"
          >
            envName
          </span>
          <span
            class="value"
          >
             
            develop
             
          </span>
        </div>
      </div>
      <div
        class="key-value-list depth-3 sub "
      >
        <div
          class="title"
        >
          application
        </div>
        <div
          class="k-v-row"
        >
          <span
            class="key"
          >
            name
          </span>
          <span
            class="value"
          >
             
            app1
             
          </span>
        </div>
      </div>
    </div>
    <div
      class="k-v-row"
    >
      <span
        class="key"
      >
        description
      </span>
      <span
        class="value"
      >
         
        this is description
         
      </span>
    </div>
  </div>
</div>
`;
